include/librealsense2/dds/dds-device.cpp

```diff
@@ -36,7 +74,180 @@ class dds_device::impl
+        {
+            throw std::runtime_error( "Error creating a DDS subscriber" );
+        }
+
+        // CREATE TOPIC TYPE
+        eprosima::fastdds::dds::TypeSupport topic_type(
+            new librealsense::dds::topics::device::notification::type );
+
+        // REGISTER THE TYPE
+        DDS_API_CALL( topic_type.register_type( _participant->get() ) );
+
+
+
+        std::string topic_name
+            = librealsense::dds::topics::device::notification::construct_topic_name( _info.topic_root );
+
+        // CREATE THE TOPIC
+        _topic = DDS_API_CALL( _participant->get()->create_topic( topic_name,
+                                                                  topic_type->getName(),
+                                                                  TOPIC_QOS_DEFAULT ) );
+
+        // CREATE THE READER
+        DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
+        rqos.data_sharing().off();
+        rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
+        rqos.durability().kind = VOLATILE_DURABILITY_QOS;
+        rqos.history().kind = KEEP_LAST_HISTORY_QOS;
+        rqos.history().depth = 10;
+        rqos.endpoint().history_memory_policy
+            = eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
+        _reader = DDS_API_CALL( _subscriber->create_datareader( _topic, rqos, nullptr ) );
+        if( _reader == nullptr )
+        {
+            throw std::runtime_error( "Error creating a DDS reader for " + topic_name );
+        }
+
+        LOG_DEBUG( "topic '" << topic_name << "' created" );
+    }
+
+    bool init()
+    {
+        size_t num_of_sensors = 0;
+        std::unordered_map< int , std::string> sensor_index_to_name;
+        std::unordered_map< std::string, topics::device::notification::video_stream_profiles_msg > sensor_to_video_profiles;
+        std::unordered_map< std::string, topics::device::notification::motion_stream_profiles_msg > sensor_to_motion_profiles;
+
+        // We expect to receive all of the sensors data under a timeout
+        utilities::time::timer t( std::chrono::seconds( 30) ); // TODO: refine time out
+        state_type state = state_type::WAIT_FOR_DEVICE_HEADER;
+
+        while( ! t.has_expired() && state_type::DONE != state )
+        {
+            if( _reader->wait_for_unread_message( { 1, 0 } ) )
+            {
+                dds::topics::raw::device::notification raw_data;
+                topics::device::notification data;
+                SampleInfo info;
+                // Process all the samples until no one is returned,
+                // We will distinguish info change vs new data by validating using `valid_data`
+                // field
+                while( ReturnCode_t::RETCODE_OK == _reader->take_next_sample( &raw_data, &info ) )
+                {
+                    // Only samples for which valid_data is true should be accessed
+                    // valid_data indicates that the instance is still ALIVE and the `take` return
```

Comment:
Instance is still alive? So... if it's not alive, shouldn't we just stop everything?
And the 'take' return? Typo?

Reply:
`valid_data == true` indicates updated sample values, `info.instance_state ==ALIVE_INSTANCE_STATE` indicates there are live writers and that the writer did not dispose of this instance.

Reply:
I took this comment from eProsima documentation.
But I liked Ohad description, I will update the comment to indicate this and remove the `alive` thing.